include/tensor.cuh

```diff
@@ -1225,8 +1254,14 @@ private:
      */
     void computeWorkspaceSize(size_t m, size_t n);
 
+
 public:
 
+    int getStatus() {
+        // redefinition of getStatus
+        return m_svdStatus;
```

Comment:
why not have `m_svdStatus = std::max(m_svdStatus, (*m_info)(0, 0, 0));` here, so you don't need the `#ifdef GPUTILS_DEBUG_MODE` blocks ?

Reply:
I've changed the code slightly:

1. In `CholeskyBatchFactoriser` we need to allocate more memory for the device info; this is required by cuBLAS; see API docs of  [`potrsbatched`](https://docs.nvidia.com/cuda/cusolver/index.html#cusolverdn-t-potrsbatched) 
2. In `Svd` we perform multiple calls of `cusolverDnDgesvd` in a for loop and we result `m_info`. If the maximum of all these `m_info` is zero, all is good. 

Anyway, let's talk about this on Teams.